tests/by-util/test_cp.rs

```diff
@@ -2310,9 +2310,15 @@ fn test_same_file_force_backup() {
 }
 
 /// Test for copying the contents of a FIFO as opposed to the FIFO object itself.
-#[cfg(unix)]
+#[cfg(all(unix, not(target_os = "freebsd")))]
 #[test]
 fn test_copy_contents_fifo() {
+    // TODO this test should work on FreeBSD, but the command was
+    // causing an error:
+    //
+    // cp: 'fifo' -> 'outfile': the source path is neither a regular file nor a symlink to a regular file
+    //
+    // the underling `std::fs:cp` doesn't support copying fifo on freeBSD
```

Comment:
```suggestion
    // the underlying `std::fs:cp` doesn't support copying fifo on freeBSD
```

Reply:
It should also be `std::fs::copy`, right?

Reply:
Looks like the linux and macos use `std::io::copy`. I think this is also related to an issue Sylvestre made back in 2020: https://github.com/rust-lang/rust/issues/79390